test/Npgsql.Tests/Replication/CommonReplicationTests.cs

```diff
@@ -454,6 +451,43 @@ async Task<NpgsqlLogSequenceNumber> GetCommitLsn(string valueString)
 
         #endregion
 
+        #region BugTests
+
+        [Test, IssueLink("https://github.com/npgsql/npgsql/issues/3534")]
+        public Task Bug3534()
+            => SafeReplicationTest(
+                async (slotName, _) =>
+                {
+                    await using var rc = await OpenReplicationConnectionAsync();
+                    var info = await rc.IdentifySystem();
+                    await CreateReplicationSlot(slotName);
+                    using var streamingCts = new CancellationTokenSource();
+                    rc.WalReceiverStatusInterval = TimeSpan.FromSeconds(1D);
+                    rc.WalReceiverTimeout = TimeSpan.FromSeconds(3D);
+                    await using var replicationEnumerator = StartReplication(rc, slotName, info.XLogPos, streamingCts.Token).GetAsyncEnumerator(streamingCts.Token);
+
+                    var replicationMessageTask = replicationEnumerator.MoveNextAsync();
+                    streamingCts.CancelAfter(rc.WalReceiverTimeout * 2);
+
+                    Assert.Multiple(() =>
```

Comment:
I don't think we need the multiple assertions block here (I've personally never used it) - but as you prefer.

Reply:
It helped me when debugging but I don't really care.